cubed/array_api/array_object.py

```diff
@@ -31,6 +37,55 @@ def __array__(self, dtype=None) -> np.ndarray:
     def __repr__(self):
         return f"cubed.Array<{self.name}, shape={self.shape}, dtype={self.dtype}, chunks={self.chunks}>"
 
+    def _repr_html_(self):
+        try:
+            grid = self.to_svg(size=ARRAY_SVG_SIZE)
+        except NotImplementedError:
+            grid = ""
+
+        if not math.isnan(self.nbytes):
+            nbytes = format_bytes(self.nbytes)
+            cbytes = format_bytes(math.prod(self.chunksize) * self.dtype.itemsize)
+        else:
+            nbytes = "unknown"
+            cbytes = "unknown"
+
+        return get_template("array.html.j2").render(
+            array=self,
+            grid=grid,
+            nbytes=nbytes,
+            cbytes=cbytes,
+            arrs_in_plan=f"{self.plan.dag.number_of_nodes()} arrays in Plan",
+            arrtype="np.ndarray",
+        )
+
+    def to_svg(self, size=500):
+        """Convert chunks from Cubed Array into an SVG Image
+
+        Parameters
+        ----------
+        chunks: tuple
+        size: int
+            Rough size of the image
+
+        Examples
+        --------
+        >>> x.to_svg(size=500)  # doctest: +SKIP
+
+        Returns
+        -------
+        text: An svg string depicting the array as a grid of chunks
+        """
+        from cubed.vendor.dask.array.svg import svg
+
+        return svg(self.chunks, size=size)
+
+    def _repr_inline_(self, max_width):
+        """
+        Format to a single line with at most max_width characters. Used by xarray.
+        """
+        return f"cubed.Array<chunksize={self.chunksize}>"
```

Comment:
Makes xarray's repr slightly better when collapsed, see https://docs.xarray.dev/en/stable/internals/duck-arrays-integration.html

Reply:
(Doesn't use the `max_width` kwarg but it's included for compatibility)